plugins/inputs/procstat/native_finder.go

```diff
@@ -82,31 +82,31 @@ func (pg *NativeFinder) FullPattern(pattern string) ([]PID, error) {
 
 // ChildPattern matches children pids on the command line when the process was executed
 func (pg *NativeFinder) ChildPattern(pattern string) ([]PID, error) {
-	var parentpids []PID
-	var pids []PID
 	regxPattern, err := regexp.Compile(pattern)
 	if err != nil {
-		return pids, err
+		return nil, err
 	}
+
 	procs, err := process.Processes()
 	if err != nil {
-		return pids, err
+		return nil, err
 	}
+
+	var pids []PID
 	for _, p := range procs {
 		cmd, err := p.Cmdline()
-		if err != nil {
+		if err != nil || !regxPattern.MatchString(cmd) {
 			continue
 		}
-		if regxPattern.MatchString(cmd) {
-			parentpids = append(parentpids, PID(p.Pid))
+
+		parent, err := process.NewProcess(p.Pid)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get process %d: %w", p.Pid, err)
 		}
-	}
-	for _, p := range parentpids {
-		parent, err := process.NewProcess(int32(p))
+		children, err := parent.Children()
 		if err != nil {
-			return pids, err
+			return nil, fmt.Errorf("unable to get children of process %d: %w", p.Pid, err)
```

Comment:
The changes here will result in no process information being returned if any of the child processes may have an exception. 
Is this the expected strategy? Will there be situations where it is necessary to retain the ability to obtain the pid?

Reply:
Do we really _use_ the PIDs in case of an error? I don't think so and if so we should not. The returned data is inconsistent and a partial view and we cannot indicate this to the server. If there is a consistent way to handle the errors we should do it in this function and only log an error but continue gathering...

This being said, I think we should not use the PIDs in case of error and thus should return `nil`. Does that make sense to you?